torch/mps/__init__.py

```diff
@@ -0,0 +1,88 @@
+r"""
+This unit enables an interface for accessing MPS backend in python
+"""
+import torch
+from functools import lru_cache
+import threading
+from .. import Tensor
+import torch._C
+
+_initialized = False
+_initialization_lock = threading.Lock()
+default_mps_generator: torch._C.Generator = None  # type: ignore[assignment]
+
+def init():
+    r"""Initialize PyTorch's MPS state.
+    Does nothing if the MPS state is already initialized.
+    """
+    _lazy_init()
+
+def _lazy_init():
```

Comment:
Why are you doing lazy initialization for this?

Reply:
The `lazy_init` is meant to initialize only "on-demand". The code is directly taken from CUDA's __init__.py:
https://github.com/pytorch/pytorch/blob/master/torch/cuda/__init__.py#L206

Reply:
But that is because cuda initialization is actually really expensive AND we don't initialize anything in cuda at import time (because cuda init allocates a couple GB in RAM and a couple 100s MB on the GPU).
My understanding here is that the only part that is lazy here is the MPS generator creation so:
- What happens if I just call `torch.rand(10, device="mps")`? This won't trigger this lazy init but actually use the mps generator?
- If initializing this generator is cheap (which I assume it is), I don't think it's worth the complexity.

Reply:
Updated the PR.